File: LeetCode/java/1684_Count_the_Number_of_Consistent_Strings.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        int answer = 0;

        Set<Character> set = new HashSet<>();

        for (int i = 0; i < allowed.length(); i++) {
            set.add(allowed.charAt(i));
        }

        for (String word : words) {
            boolean flag = false;

            for (int i = 0; i < word.length(); i++) {
                if (!set.contains(word.charAt(i))) {
                    flag = true;
                    break;
                }
            }

            if (!flag) {
                answer++;
            }
        }

        return answer;
    }
}

File: LeetCode/cpp/1684_Count_the_Number_of_Consistent_Strings.cpp
class Solution {
public:
	int countConsistentStrings(string allowed, vector<string>& words) {
		int answer = 0;

		set<char> st;

		for (int i = 0; i < allowed.size(); i++) {
			st.insert(allowed[i]);
		}

		for (string word : words) {
			bool flag = false;

			for (int i = 0; i < word.size(); i++) {
				if (st.find(word[i]) == st.end()) {
					flag = true;
					break;
				}
			}

			if (!flag) {
				answer++;
			}
		}

		return answer;
	}
};